RoboJackets/igvc-software
# 3D Localization and Holonomic Teb Configuration

# Description

Adds 3D localization using the `robot_localization` package as well as a new TEB configuration for swervi's holonomic motion

This PR does the following:
- 3D localization using the `robot_localization` package
- New TEB configuration for Swervi's holonomic motion
- Tuned PID parameters for the gazebo controller
- Adjusted height of Swervi so it can actually go over ramps instead of getting stuck on the hump <sup>lol</sup>

Fixes #815 and Fixes #662

# Testing steps
## Test Case 1
1. Run `roslaunch igvc_gazebo ramp_lane.launch`
2. Run `roslaunch igvc_navigation localization.launch`
3. Open RViz
4. Move Swervi to the peak of the ramp

Expectation: The /odometry/filtered pose should reflect the height and orientation of the robot on the ramp
![image](https://user-images.githubusercontent.com/24273822/166398576-57dca0f7-4c1c-48d4-9906-a4678c9df7e7.png)

## Test Case 2
1. Run `roslaunch igvc_gazebo autonav_low.launch`
2. Run `roslaunch igvc_navigation navigation_simulation.launch`

Expectation: Observe that Swervi is able to navigate while moving sideways (not only forward).

# Self Checklist
